### DISCUSSION TITLE
Temporary File Upload to S3 returns the S3 URL instead of what is configured on the environment file

### DISCUSSION THREAD

**Discussion by abishekrsrikaanth** [livewire/livewire#4700] [7. Bugs (archived)]
When using the `S3` as the filesystem for `temporary_file_upload.disk`, the returned temporary URL is actually the S3 URL and not the one configured on `AWS_URL`. Currently, we have S3 public access restricted and assets are only available through Cloudfront, which is one of the reasons why Laravel provides the `AWS_URL` setting on the `filesystems.php` config. 

I currently have to replace the S3 URL with CloudFront URL
```php

$temporaryUrl = $this->file->temporaryUrl();

$newTemporaryUrl = str_replace(
            parse_url($url)['host'],
            parse_url(config('filesystems.disks.s3.url'))['host'],
            $url
        );
```

Is there any way to have livewire return the URL configured on `AWS_URL` instead of the actual S3 URL?

**Comment by joshhanley** [comment]
@abishekrsrikaanth I don't think there is currently any way to do it. But I'm sure a PR that adds support for this would be welcome
